jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/UserConfigurationCompareTest.java

```diff
@@ -0,0 +1,158 @@
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+public class UserConfigurationCompareTest {
+
+    @Test
+    public void TestConfigEqualWhenEqual() {
+        //Given
+        Object entity1 = "Hello";
+        Object entity2 = "Hello";
+        //When
+        boolean result = UserConfigurationCompare.configEqual(entity1, entity2);
+        //Then
+        assertTrue(result);
+    }
+
+    @Test
+    public void TestConfigEqualWhenEitherNull() {
+        //Given
+        Object entity1 = null;
+        Object entity2 = "new";
+        //When
+        boolean result = UserConfigurationCompare.configEqual(entity1, entity2);
+        //Then
+        assertFalse(result);
+    }
+
+    @Test
+    public void TestConfigEqualWhenMapAndEqual() {
+        //Given
+        Map<String, String> entity1 = new HashMap();
+        Map<String, String> entity2 = new HashMap();
+        entity1.put("item1", "code");
+        entity2.put("item1", "code");
+        //When
+        boolean result = UserConfigurationCompare.configEqual(entity1, entity2);
+        //Then
+        assertTrue(result);
+    }
+    @Test
+    public void TestConfigEqualWhenMapAndNotEqual() {
+        //Given
+        Map<String, String> entity1 = new HashMap();
+        Map<String, String> entity2 = new HashMap();
+        entity1.put("item", "code");
+        entity2.put("item1", "code");
+        //When
+        boolean result = UserConfigurationCompare.configEqual(entity1, entity2);
+        //Then
+        assertFalse(result);
+    }
+
+
+    @Test
+    public void TestConfigEqualWhenObjectMetaIsTrue() {
+        //Given
+        Object entity1 = new ObjectMetaBuilder().withName("test1").withAnnotations(Collections.singletonMap("foo", "bar")).build();
+        Object entity2 = new ObjectMetaBuilder().withName("test1").withAnnotations(Collections.singletonMap("foo","bar")).build();
+        //When
+        boolean result = UserConfigurationCompare.configEqual(entity1, entity2);
+        //Then
+        assertTrue(result);
+    }
+
+    @Test
+    public void TestConfigEqualWhenCollectionWhenFalse(){
+        //Given
+        ArrayList<String> anandList = new ArrayList<>();
+        Object entity1 = Collections.singletonList(anandList);
+        Object entity2 = Collections.EMPTY_LIST;
+        //When
+        boolean result = UserConfigurationCompare.configEqual(entity1,entity2);
+        //Then
+        assertFalse(result);
+    }
+
+    @Test
+    public void TestConfigEqualWhenCollectionWhenTrue(){
+        //Given
+        Object entity1 = Collections.EMPTY_LIST;
+        Object entity2 = Collections.EMPTY_SET;
+        //When
+        boolean result = UserConfigurationCompare.configEqual(entity1,entity2);
+        //Then
+        assertTrue(result);
+    }
+
+
+    @Test
+    public void TestConfigEqual() {
```

Comment:
nit, here test name is `TestConfigEqual` but you're comparing two non-equal entities. Is this okay?

Reply:
Also remember this is not Go, methods **never** start with capital/uppercase letters since it will confuse readers of your code who will understand you are referencing a class wherever you call that method (this is not so bad for a test case method, but we should try to adhere to conventions).